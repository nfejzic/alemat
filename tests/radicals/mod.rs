use alemat::{
    elements::{radicals::Radical, scripted::SubSup, Ident, Num, Operator},
    MathMl, MathMlFormatter,
};

#[test]
fn sqrt() {
    let output = MathMl::with_content(
        Radical::builder()
            .index("2")
            .content(alemat::children![
                Num::from(1),
                Operator::from("+"),
                SubSup::builder()
                    .base(Ident::from("n"))
                    .subscript(Num::from(2))
                    .supscript(Num::from(3))
                    .build(),
            ])
            .build(),
    )
    .render_with(&mut MathMlFormatter);

    crate::snap_test!(output, name: "radicals_sqrt");
}

#[test]
fn root() {
    let output = MathMl::with_content(
        Radical::builder()
            .index("3")
            .content(alemat::children![
                Num::from(1),
                Operator::from("+"),
                SubSup::builder()
                    .base(Ident::from("n"))
                    .subscript(Num::from(2))
                    .supscript(Num::from(3))
                    .build(),
            ])
            .build(),
    )
    .render_with(&mut MathMlFormatter);

    crate::snap_test!(output, name: "radicals_root");
}
