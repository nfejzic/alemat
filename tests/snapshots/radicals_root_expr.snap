---
source: tests/radicals/mod.rs
expression: input
---
<math>
  <mroot>
    <mrow>
      <mn>
        1
      </mn>
      <mo>
        +
      </mo>
      <msubsup>
        <mi>
          n
        </mi>
        <mn>
          2
        </mn>
        <mn>
          3
        </mn>
      </msubsup>
    </mrow>
    <mrow>
      <mi>
        n
      </mi>
      <mo>
        +
      </mo>
      <mn>
        1
      </mn>
    </mrow>
  </mroot>
</math>

